xfrd.c

```diff
@@ -1965,16 +1986,15 @@ xfrd_parse_received_xfr_packet(xfrd_zone_type* zone, buffer_type* packet,
 				zone->apex_str, zone->master->ip_address_spec));
 			region_destroy(tempregion);
 			return xfrd_packet_bad;
-		}
-		if(zone->soa_disk_acquired != 0 && zone->soa_disk.serial == soa->serial) {
-			DEBUG(DEBUG_XFRD,1, (LOG_INFO, "xfrd: zone %s got "
-						       "update indicating "
-						       "current serial",
+		} else if(compare_serial(ntohl(soa->serial),
+		                         ntohl(last_soa->serial)) == 0)
+		{
+			DEBUG(DEBUG_XFRD,1, (LOG_INFO,
+				"xfrd: zone %s got update indicating current serial",
 				zone->apex_str));
 			/* (even if notified) the lease on the current soa is renewed */
-			zone->soa_disk_acquired = xfrd_time();
-			if(zone->soa_nsd.serial == soa->serial)
-				zone->soa_nsd_acquired = xfrd_time();
+			*last_soa_acquired = xfrd_time();
+			/* FIXME: zone state should not be updated here(?) */
```

Comment:
Yes the zone state should be updated because we got a new transfer indicating the serial.  Hence we got a new lease time for the serial and thus the timestamp that denotes the leasetime is updated and also the zone state can be changed from (whatever, expired, retrying) to ok.  This means the FIXME comment can be removed, I think?

Reply:
And also the removed line that updates the soa_nsd_acquired if it also needs updating for the timestamp is needed.  Because the zone may be loaded and that zone has a new lease, is more recent, right?

Reply:
Better safe than sorry I guess. Removed the comment. As for "the removed line": `last_soa_acquired` points to either the acquired for the transfer or the one that's loaded. Depending on whether there's a transfer or not.